@ndlib/gatsby-theme-marble/src/components/Internal/Seo/index.js

```diff
@@ -32,20 +32,25 @@ export const Seo = ({
       }
     `
   )
-  const frontmatter = typy(data, 'remarkMarblePage.frontmatter').safeObject
   const { siteMetadata } = site
+  const itemTitle = title || typy(data, 'marbleItem.title').safeString
+  const itemDescription = description || typy(data, 'marbleItem.description').safeString
+  const itemImage = typy(data, 'marbleItem.childrenMarbleFile[0].iiif.thumbnail').safeString
+  const creator = typy(data, 'marbleItem.metadata').safeArray.find(md => {
+    return md.label === 'Creator'
+  })
+  const itemAuthor = author || typy(creator, 'value[0]').safeString
```

Comment:
I'd like to see this logic moved to the `getTitle`, `getAuthor` and `getDescription` functions so that all the work to determine a particular field is contained within a single function instead of splitting the logic into the first two cases and a third possibility. So basically, `getTitle` for example would become:
```
export const getTitle = (title, marbleItem, siteMetadata) => {
  return title || typy(data, 'marbleItem.title').safeString || siteMetadata.title
}
```

Reply:
Refactored component and modified unit tests to this approach